File: 프로그래머스/lv2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.cpp
#include <vector>

using namespace std;
#include <iostream>
vector<long long> solution(vector<long long> numbers) {
    vector<long long> answer;
    for (long long num : numbers) {
        long long bit = 1;
        while ((num & bit) == bit) bit <<= 1;
        num |= bit; 
        answer.push_back((num & (~(bit >> 1))));
    }
    return answer;
}

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.java
import java.util.*;

class Solution {
    
    private long f(long x) {
        long bit = 1;
        while ((x & bit) == bit) bit <<= 1;
        x |= bit;
        return x & ~(bit >> 1);
    }
    
    public long[] solution(long[] numbers) {
        return Arrays.stream(numbers).map(this::f).toArray();
    }
}